packages/uniforms/__tests__/AutoForm.tsx

```diff
@@ -144,51 +152,90 @@ describe('<AutoForm />', () => {
 
   describe('when reset', () => {
     it('reset `model`', () => {
-      const Component = () => (
-        <AutoForm autosave model={model} schema={schema}>
+      const schema = new SimpleSchema({ a: { type: String, optional: true } });
+      const bridge = new SimpleSchema2Bridge(schema);
+      render(
+        <AutoForm schema={bridge}>
           <context.Consumer children={contextSpy} />
-        </AutoForm>
+          <AutoFields />
+        </AutoForm>,
+      );
+
+      expect(contextSpy).toHaveBeenLastCalledWith(
+        expect.objectContaining({ model: {} }),
+      );
+
+      const input = screen.getByLabelText('A');
+      fireEvent.change(input, { target: { value: 'a' } });
+
+      expect(contextSpy).toHaveBeenLastCalledWith(
+        expect.objectContaining({ model: { a: 'a' } }),
       );
-      const { rerender } = render(<Component />, schemaDefinition);
 
-      rerender(<Component />);
+      contextSpy.mock.calls.slice(-1)[0][0]?.formRef.reset();
```

Comment:
Wow, don't we want to have a reusable function to reset the contextSpy? In this form it looks terrifying

Reply:
Now that I look at it... 😅 I'll improve it in the next PRs.